### DISCUSSION TITLE
412 (Precondition Failed)

### DISCUSSION THREAD

**Discussion by MaxenceIPSSI** [emailjs-com/help#110] [Q&A] [answered]
Hello,
I'm a french student working for the first time on a big e-market website and I have a little issue to send my form by mail with EmailJS.
This is my sending code : 
![Capture d’écran 2024-05-23 171830](https://github.com/emailjs-com/help/assets/150336308/964babc8-3d46-4dd3-a7ee-8538229557df)

"form.current" successfully get my form because when I console.log it, I get what I want to get : 
![Capture d’écran 2024-05-23 171921](https://github.com/emailjs-com/help/assets/150336308/cf488321-02b6-4763-89bb-58f76c87afc7)

Now there is my template : 
![Capture d’écran 2024-05-23 172209](https://github.com/emailjs-com/help/assets/150336308/8affd5a7-b808-4b5b-ad35-16419a9f2df3)

As you can see, I get my {{from_name}} value directly from the template, and my {{adresse_un}} value from my form.
I checked again and again my template id, service id and public key and they are all right.

However, I can't send a mail, I have these errors in console : 
![Capture d’écran 2024-05-23 172604](https://github.com/emailjs-com/help/assets/150336308/2e48f636-52a9-4655-97b7-3984b61fa040)

After some researches about 412 error, I tought understand that could be because of the information I send not matching with what EmailJS is waiting for. But as I explained, for me, everything is fine.

That's why I need your help to make me happy.
Ty for reading !

**Comment by xr0master** [comment, marked as answer]
As you can see, the issue here is the connection to Gmail. You need to allow the permission to "send email on my behalf."
